slack_bolt/app/app.py

```diff
@@ -853,6 +861,50 @@ def __call__(*args, **kwargs):
 
         return __call__
 
+    def function(
+        self,
+        callback_id: Union[str, Pattern],
+        matchers: Optional[Sequence[Callable[..., bool]]] = None,
+        middleware: Optional[Sequence[Union[Callable, Middleware]]] = None,
+    ) -> Callable[..., Optional[Callable[..., Optional[BoltResponse]]]]:
+        """Registers a new Function listener.
+        This method can be used as either a decorator or a method.
+            # Use this method as a decorator
+            @app.function("reverse")
+            def reverse_string(event: dict, client: WebClient, context: BoltContext):
+                try:
+                    string_to_reverse = event["inputs"]["stringToReverse"]
+                    client.functions_completeSuccess(
```

Comment:
Should this example use the client directly, or the `complete` / `fail` shorthand parameter callbacks?

Reply:
Good call 💯  I think `complete` / `fail` shorthand parameter are better

This example it good to  show how to hand roll completing a function 🤔 not sure where we could include it

Reply:
I think just explaining what `complete`/`fail` do, and which underlying HTTP API methods they invoke, is sufficient. Link to the relevant methods and developers will figure the rest out.